ecl/hql/hqlfold.cpp

```diff
@@ -7075,15 +7075,41 @@ class LowerCaseTransformer : public NewHqlTransformer
 
     virtual IHqlExpression * createTransformed(IHqlExpression * expr)
     {
-        switch (expr->getOperator())
+        node_operator op = expr->getOperator();
+        switch (op)
         {
         case no_constant:
             return getLowerCaseConstant(expr);
+        case no_cast:
+        case no_implicitcast:
+        case no_concat:
+        case no_nofold:
+        case no_nocombine:
+        case no_nohoist:
+        case no_alias:
+        case no_globalscope:
+        case no_map:
+        case no_trim:
+            return NewHqlTransformer::createTransformed(expr);
+        case no_if:
+        case no_mapto:
+        case no_case:
+            break;
+        default:
+            return LINK(expr);
```

Comment:
@ghalliday should consider changing the name of this the LowerCaseTransformer class to reflect the purpose  - lower-casing is now conditional.